src/components/TooltipTableCellHeader/TooltipTableCellHeader.tsx

```diff
@@ -0,0 +1,52 @@
+import { Tooltip } from "@saleor/macaw-ui";
+import React from "react";
+import { MessageDescriptor } from "react-intl";
+import { useIntl } from "react-intl";
+
+import { TableCellHeaderProps } from "../TableCellHeader";
+import { tooltipTableCellHeaderMessages as messages } from "./messages";
+import { RefTableCellHeader } from "./RefTableCellHeader";
+
+interface TooltipTableCellHeaderProps extends TableCellHeaderProps {
+  filterDependency?: string;
+  overrideMessage?: MessageDescriptor;
+}
+
+export const TooltipTableCellHeader: React.FC<TooltipTableCellHeaderProps> = props => {
+  const {
+    children,
+    filterDependency,
+    overrideMessage,
+    disabled,
+    ...rest
+  } = props;
+
+  const intl = useIntl();
+
+  const tooltipDisabled = () => {
+    if (!filterDependency && !overrideMessage) {
+      return true;
+    }
+    return !disabled;
+  };
+
+  return (
+    <Tooltip
+      title={intl.formatMessage(
+        !!overrideMessage ? overrideMessage : messages.noFilterSelected,
+        {
+          filterName: filterDependency
+        }
+      )}
+      placement="top"
+      disabled={tooltipDisabled()}
+    >
```

Comment:
Can we improve the interface a bit? I think, ideally, I should be able to say what this particular component does only based on the interface it provides like its name and props without knowing anything about the implementation and without knowing in what context it is used. `TooltipTableCellHeader` looks like generic name, while `filterDependency` looks not connected to it in any way.
I guess it would be cleaner to move message outside of component permanently. What do you think about it? Do you agree with me? 
```suggestion
  const {
    children,
    tooltip,
    disabled,
    ...rest
  } = props;

  const intl = useIntl();

  const tooltipDisabled = () => {
    if (!tooltip) {
      return true;
    }
    return !disabled;
  };

  return (
    <Tooltip
      title={tooltip}
      placement="top"
      disabled={tooltipDisabled()}
    >
```

Reply:
That sounds like a good idea, however this message is repeated across multiple lists and I don't want to duplicate it. Maybe putting it in `src/intl.ts` would be a good solution here?

Reply:
I think `src/intl.ts` is a bit too general - I think I would just put it in the same folder as this tooltip.

Reply:
Yep, imo might be in put it in the same folder as well